source/solvers/navier_stokes_vof_assemblers.cc

```diff
@@ -472,8 +461,7 @@ GLSNavierStokesVOFAssemblerBDF<dim>::assemble_rhs(
           for (unsigned int p = 0; p < number_of_previous_solutions(method) + 1;
                ++p)
             {
-              local_rhs_i -=
-                densities[p] * bdf_coefs[p] * (velocity[p] * phi_u_i);
+              local_rhs_i -= density * bdf_coefs[p] * (velocity[p] * phi_u_i);
```

Comment:
I am not sure that I understand why `densities[p]` was not the right term?

Reply:
Good question. Because we don't calculate the derivative of (rho*u) but instead rho * derivative of (u). This is generally the formulation usedin most articles. We can talk about it more if you want :)